operator/pkg/translate/translate.go

```diff
@@ -211,19 +218,7 @@ func (t *Translator) OverlayK8sSettings(yml string, iop *v1alpha1.IstioOperatorS
 		if componentName == name.IstioBaseComponentName {
 			return "", fmt.Errorf("base component can only have k8s.overlays, not other K8s settings")
 		}
-		// for server-side apply, make sure service port has protocol defined.
-		// TODO(richardwxn): remove after https://github.com/kubernetes-sigs/structured-merge-diff/issues/130 is fixed.
 		inPathParts := strings.Split(inPath, ".")
-		if inPathParts[len(inPathParts)-1] == "Service" {
```

Comment:
Why is this removed?

Reply:
This was a workaround for: https://github.com/kubernetes-sigs/structured-merge-diff/issues/130. Now it can be removed since that was fixed upstream a while ago

Reply:
This is already a complex and risky PR, can we not couple it with other changes?

I don't think this is safe to be removed, as it requires the api-server to be updated so we will break old versions?